File: aoc_2019_python/day05.py
import aoc
from intcode import Intcode

def main(puzzle_input):
    computer = Intcode(puzzle_input)
    computer.input = [1]
    computer.run_program()
    part1 = computer.output[-1]

    computer = Intcode(puzzle_input)
    computer.input = [5]
    computer.run_program()
    part2 = computer.output[0]

    return part1, part2
 
aoc.run_raw(main, "day05.txt")

File: aoc_2018_java/src/Day05.java
public class Day05 implements AocSolver {
	public static void main(String[] args) {
		Aoc.runRaw(new Day05());
	}

	@Override
	public void solve(String input) {
		int part1 = react(input).length();
		
		Integer bestReaction = null;
		for (int ascii = (int) 'a'; ascii <= (int) 'z'; ascii++) {
			String c = "" + (char) ascii;
			String polymer = input;
			polymer = polymer.replace(c, "");
			polymer = polymer.replace(c.toUpperCase(), "");
			polymer = react(polymer);
			if (bestReaction == null || polymer.length() < bestReaction) {
				bestReaction = polymer.length();
			}
		}
		int part2 = bestReaction;

		System.out.println("Part 1: " + part1);
		System.out.println("Part 2: " + part2);
	}
	
	private String react(String polymers) {
		int size = 0;
		while (size != polymers.length()) {
			size = polymers.length();
			for (int ascii = (int) 'a'; ascii <= (int) 'z'; ascii++) {
				String c = "" + (char) ascii;
				polymers = polymers.replace(c + c.toUpperCase(), "");
				polymers = polymers.replace(c.toUpperCase() + c, "");
			}
		}
		return polymers.trim();
	}
}